src/DynamoManipulation/NodeManipulator.cs

```diff
@@ -88,6 +88,28 @@ internal IRenderPackageFactory RenderPackageFactory
         }
 
         internal Point3D? CameraPosition { get; private set; }
+        
+        // This is null for each new manipulator that is initialized 
+        private bool? enabled;
+
+        /// <summary>
+        /// A manipulator is enabled when its corresponding node is
+        /// not frozen, its geometry preview is visible and its cached value is valid.
+        /// </summary>
+        public bool Enabled
+        {
+            get 
+            {
+                if (enabled == null)
```

Comment:
I'm confused by this - so once this is cached, it will never be recomputed? Shouldn't this backing field get set to null everywhere you need a recomputation to occur? Like when the cachedvalue is modified?

Reply:
It is set to null whenever a new manipulator is instantiated.

Reply:
sorry, I'm still confused here - does a new manipulator get created whenever the node gets executed? Or is the old manipulator potentially reused from a past execution?

Reply:
An old manipulator is reused from a past execution (i.e. even if point node is updated). A new manipulator is created each time you select a new node or reselect an existing node.